### PR TITLE
Week3ticktactoe

### PR DESCRIPTION
**Pull request by Christopher-Willis**
## Checkpoint Rubric
This is the rubric that your instructor will use to grade your checkpoints. Please do not edit. 

**Checkpoint 1**
- [ ] All tests passed: 40 points
- [ ] Proper use of documentation (commenting on code): 15 points
- [ ] Properly indented code: 15 points
- [ ] Demonstrated effective use of JavaScript: 30 points

**Checkpoint 2**
- [ ] The application works as it should: 40 points
- [ ] Proper use of documentation (commenting on code): 15 points
- [ ] Properly indented code: 15 points
- [ ] Demonstrated effective use of JavaScript and the DOM API: 30 points

**Checkpoint 3**
- [ ] Use of React: 25 points
- [ ] Accesses an API: 25 points
- [ ] Proper use of documentation (commenting on code): 25 points
- [ ] The application functions as it should: 25 points

### CODE REVIEW

File: 03week/ticTacToe.js [hunk 1]
```diff
@@ -14,36 +18,155 @@ let board = [
...
+const switchPlayer = (playerTurn) => {
+  if(playerTurn == "X"){
+    return "O"
+  }
+  return "X"
+}
+
+
+const ticTacToe = (row, column) => {
+  const formatedRow = parseInt(row);
```
Comment (ericbarrs) [inline 03week/ticTacToe.js:132]: intParse is a great to control converting the number however you need to control whether i can input a decimal, letter, or negative

File: 03week/ticTacToe.js [hunk 2]
```diff
@@ -14,36 +18,155 @@ let board = [
...
+  const formatedColumn = parseInt(column);
+  if (whatIsAtPosition(formatedRow,formatedColumn)!= "err" && !whatIsAtPosition(formatedRow,formatedColumn).trim()){
+    // strange big of not logic. Basically, if you don't get an error code and
+    // the spot is blank do this block. The trim is because our blanks are technically
+    // spaces. 
+    board[formatedRow][formatedColumn] = playerTurn;
+    if (checkForWin()){
```
Comment (ericbarrs) [inline 03week/ticTacToe.js]: game should show the last move before the reset is triggered.

File: 03week/ticTacToe.js [hunk 3]
```diff
@@ -14,36 +18,155 @@ let board = [
...
+    // I don't like the way I sloved this, likely to refactor
+  }
 }
 
-function verticalWin() {
-  // Your code here
+const diagonalWin = () => {
+  //left to right diaginal
+  let stringToArrayHolder = "";
+  //another string to array builder
```
Comment (ericbarrs) [inline 03week/ticTacToe.js:78]: great use of manipulating the board array.

**Comment by Christopher-Willis** [comment]
Corrections made, added a validator to simplify code, the code actually
worked before because the whatIsAtPosition function had all my error
correcting. However, in parts of my code I wasn't passing in the formatted
(parceInted) values but the regular row,column values so it could blow up
sometimes. Those errors are corrected now. Code is cleaner, and works
perfectly now. The validator isn't dynamically scalable, but I am done with
Tic Tac Toe for now :D

Pull request to regrade

https://github.com/BeCurieUs/javascript-workbook/pull/6

**Pull request merged**